### DISCUSSION TITLE
Error auto incrementing column on create

### DISCUSSION THREAD

**Discussion by ReyJust** [prisma/prisma#21675] [Q&A] [answered]
### Question

Hello!

I have a small issue here, if you guys think it is a bug, then I'll create a proper github issue.

After resetting my database and then populating it, I tried to create a new row (with `Prisma.table_name.create()`) in one of my tables and received a error:
```
Unique constraint failed on the fields: (`id`)
```

I have diagnosed that the issue occurs because I inserted beforehand a row without Prisma (in sql). Prisma failed to understand that there is already a row with id '1' and create that error. On the second try, Prisma then work as expected, by creating a row with id '2'.

I noticed that If I initially insert 2 rows instead of 1, Prisma will have the same problematic behaviors for the first 2 rows this time.

What I found curious is that, I have another table which is nearly identical (at least the auto incremental part) which doesn't have this issue. In fact I initially have 5 rows in that table, and Prisma understand that the next creation will be on id 6. Furthermore i compared both tables on DBeaver and found that the autoincrement is the same for both table. I don't think that the problem rely in the database schema. For both tables, the id is generated by ` id               Int               @id() @default(autoincrement())` in the schema.prisma

Additionally, a `prisma.table_name.count()` just before the create query gives the correct table length.

In development, I usually reset my database with `prisma migrate reset --schema=schema.prisma` and populating it with `PGPASSWORD=${DB_PASSWORD} psql -h ${DB_HOST} -p ${PORT} -U ${DB_USERNAME} -w -d ${DB_NAME} -a -f migration.sql`.
`migration.sql` contain simple raw SQL query such as `COPY [...] FROM [...].csv` which insert the inital rows.

Thanks for anyone who can help me!

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: Ubuntu 22.10
- Database: psql (PostgreSQL) 14.8 (Debian 14.8-1.pgdg120+1)
- Node.js version: v14.20.1

- Run `prisma -v` to see your Prisma version and paste it
4.16.2

**Comment by ludralph** [comment]
Hi @ReyJust 👋 

Thank you for raising this question.

I tried reproducing the problem by creating a simple table using Prisma. The model for the table is:
```prisma
model Post{
  id Int @id @default(autoincrement())
  title String
  content String
  published Boolean @default(false)
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}
```
Then i ran the command `npx prisma db push` to create. After that i opened `psql` command line interface and wrote a simple SQL to insert records into the table. 
```sql
INSERT INTO "Post" (title, content, published, "createdAt", "updatedAt") VALUES ('Your Title', 'Your Content', true, NOW(), NOW());
```
This was inserted correctly. I then switched to Prisma and wrote a simple `create` statement as shown below:
```js
 const res = await prisma.post.create({
  data:{
    title: "test",
    content: "test",
    published: true,
    createdAt: new Date(),
    updatedAt: new Date(),
  }
 })
```
This was also inserted correctly and the autoincrement worked as expected. I also inserted 2 rows instead of 1. Prisma was also able to correctly infer the next autoincremented value without any issues. 

Can you please share the query you are running in your `migration.sql` file so I can try reproduce that exact scenario to debug this further.  Also, it would be helpful if you can share the [logs](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging#log-to-stdout) from when the `autoincrement` fail so we can understand better what's going on under the hood.

**Comment by ReyJust** [comment]
Prisma logs gave me:
```
prisma:query BEGIN
prisma:query DEALLOCATE ALL
prisma:query INSERT INTO "public"."filters" ("name","created_at","updated_at","tenant_id","properties","description","custom","active","isNumeric","isUserNumeric") VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10) RETURNING "public"."filters"."id"
prisma:query ROLLBACK
```
I tried to run the query by myself in Dbeaver like so:
```
INSERT INTO "public"."filters" ("name","created_at","updated_at","tenant_id","properties","description","custom","active","isNumeric","isUserNumeric") VALUES (
'a','2021-01-01','2021-01-01','a','{}','a',true,true,true,true
) RETURNING "public"."filters"."id"```

and I was able to reproduce the error. Would that mean that the error isn't originated by prisma?

**Comment by ReyJust** [comment, marked as answer]
I think this explain my issue https://github.com/prisma/prisma/discussions/5256#discussioncomment-307754 . Nothing related to Prisma, I just need to remove the "ids" from my mock data files. Thanks!!
